command/agent/retry_join.go

```diff
@@ -187,7 +187,8 @@ func (r *retryJoiner) RetryJoin() {
 		}
 
 		if len(addrs) > 0 && r.joinFunc != nil {
-			numJoined, err := r.joinFunc(addrs)
+			var numJoined int
+			numJoined, err = r.joinFunc(addrs)
```

Comment:
Shouldn't the `if err != nil` check down on L205 be above the retry check on L198-203? Seems like the last error encountered still wouldn't get logged.